src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/RazorSyntaxTreeExtensions.cs

```diff
@@ -24,7 +24,7 @@ private static ImmutableArray<RazorDirectiveSyntax> GetDirectives(RazorSyntaxTre
     {
         using var builder = new PooledArrayBuilder<RazorDirectiveSyntax>();
 
-        foreach (var node in syntaxTree.Root.DescendantNodes(ShouldDescendIntoChildren))
+        foreach (var node in syntaxTree.Root.DescendantNodes(MayContainDirectives))
```

Comment:
Could consider an `EnumerateDirectives` extension method in this class, which is used here and in the folding range provider.